protocol/transaction.go

```diff
@@ -0,0 +1,40 @@
+package protocol
+
```

Comment:
should we move this file content to `extend.go`? 🤔

Reply:
we could. it didn't seem fit to me to talk the concepts used in posgres documentation of the flow and not in the reference. the actual technical term for this is transaction. it is also mentioned in the docs